cpp/modmesh/toggle/RadixTree.hpp

```diff
@@ -68,66 +68,59 @@ class TimedEntry
 template <typename T>
 class RadixTreeNode
 {
-private:
-    using ChildList = std::list<std::unique_ptr<RadixTreeNode>>;
-
 public:
-    RadixTreeNode()
-        : prev(nullptr){};
-    RadixTreeNode(std::string name, int key)
-        : m_name(std::move(name))
+
+    using child_list_type = std::list<std::unique_ptr<RadixTreeNode<T>>>;
+    using key_type = int32_t;
+
+    static_assert(std::is_integral_v<key_type> && std::is_signed_v<key_type>, "signed integral required");
+
+    RadixTreeNode(std::string const & name, key_type key)
+        : m_name(name)
         , m_key(key)
-        , prev(nullptr)
+        , m_prev(nullptr)
     {
     }
 
-    const int & getKey() const { return m_key; }
-    const std::string & getName() const { return m_name; }
-    T & getData() { return m_data; }
-    const T & getData() const { return m_data; }
-    const ChildList & getChildren() const { return m_children; }
+    RadixTreeNode() = default;
+    RadixTreeNode(RadixTreeNode const &) = default;
+    RadixTreeNode(RadixTreeNode &&) = default;
+    RadixTreeNode & operator=(RadixTreeNode const &) = default;
+    RadixTreeNode & operator=(RadixTreeNode &&) = default;
+    ~RadixTreeNode() = default;
+
+    key_type key() const { return m_key; }
+    const std::string & name() const { return m_name; }
+    T & data() { return m_data; }
+    const T & data() const { return m_data; }
+    const child_list_type & children() const { return m_children; }
 
-    // Add a child node
-    RadixTreeNode<T> * addChild(std::string childName, int childKey)
+    RadixTreeNode<T> * add_child(std::string const & name, key_type key)
     {
-        auto newChild = std::make_unique<RadixTreeNode<T>>(std::move(childName), std::move(childKey));
-        newChild->prev = this;
-        m_children.push_back(std::move(newChild));
+        auto new_child = std::make_unique<RadixTreeNode<T>>(name, key);
+        new_child->m_prev = this;
+        m_children.push_back(std::move(new_child));
         return m_children.back().get();
     }
 
-    // Get a child node with a given key
-    RadixTreeNode<T> * getChild(int key) const
+    RadixTreeNode<T> * get_child(key_type key) const
     {
         auto it = std::find_if(m_children.begin(), m_children.end(), [&](const auto & child)
-                               { return child->getKey() == key; });
+                               { return child->key() == key; });
         return (it != m_children.end()) ? it->get() : nullptr;
     }
 
-    // Get prev node
-    RadixTreeNode<T> * getPrev() const
+    RadixTreeNode<T> * get_prev() const
```

Comment:
Is single-name accessor style also applicable to this?

Reply:
I haven't decided, so leave it as a getter.  `get_prev` and `get_child` look like a group so that I keep them in the same style.  And I am not yet sure about whether a pointer or a reference should be returned.  It makes more sense to return a reference with the single-name style.

Let's keep it as is for a while.